plugins/woocommerce/includes/gateways/paypal/includes/class-wc-gateway-paypal-request.php

```diff
@@ -254,40 +277,62 @@ public function capture_authorized_payment( $order ) {
 		}
 
 		// Skip if the payment is already captured.
-		if ( 'completed' === $order->get_meta( '_paypal_status', true ) ) {
+		if ( 'captured' === $order->get_meta( '_paypal_status', true ) || 'completed' === $order->get_meta( '_paypal_status', true ) ) {
 			WC_Gateway_Paypal::log( 'PayPal payment is already captured. Skipping capture. Order ID: ' . $order->get_id() );
 			return;
 		}
 
-		$request_url = $this->get_paypal_capture_authorized_payment_request_url( $order->get_transaction_id() );
+		$paypal_debug_id = null;
+		try {
+			$request_url = $this->get_paypal_capture_authorized_payment_request_url( $order->get_transaction_id() );
 
-		$response = wp_remote_post(
-			$request_url,
-			array(
-				'method'  => 'POST',
-				'headers' => array(
-					'Content-Type' => 'application/json',
-				),
-				'body'    => wp_json_encode( array( 'testmode' => $this->gateway->testmode ) ),
-				'timeout' => 60,
-			)
-		);
+			$response = wp_remote_post(
+				$request_url,
+				array(
+					'method'  => 'POST',
+					'headers' => array(
+						'Content-Type' => 'application/json',
+					),
+					'body'    => wp_json_encode( array( 'testmode' => $this->gateway->testmode ) ),
+					'timeout' => 60,
+				)
+			);
 
-		if ( is_wp_error( $response ) ) {
-			WC_Gateway_Paypal::log( 'PayPal capture payment request failed. Response error: ' . $response->get_error_message() );
-			return;
-		}
+			if ( is_wp_error( $response ) ) {
+				throw new Exception( 'PayPal capture authorizedpayment failed. Response error: ' . $response->get_error_message() );
+			}
 
-		$http_code     = wp_remote_retrieve_response_code( $response );
-		$body          = wp_remote_retrieve_body( $response );
-		$response_data = json_decode( $body, true );
+			$http_code     = wp_remote_retrieve_response_code( $response );
+			$body          = wp_remote_retrieve_body( $response );
+			$response_data = json_decode( $body, true );
 
-		if ( 200 !== $http_code ) {
-			WC_Gateway_Paypal::log( 'PayPal capture payment failed. Response status: ' . $http_code . '. Response body: ' . $body );
-		}
+			if ( 200 !== $http_code ) {
+				$paypal_debug_id = isset( $response_data['debug_id'] ) ? $response_data['debug_id'] : null;
+				throw new Exception( 'PayPal capture authorized payment failed. Response status: ' . $http_code . '. Response body: ' . $body );
+			}
+
+			// set custom status for successful capture response.
+			$order->update_meta_data( '_paypal_status', 'captured' );
```

Comment:
The status is `$response_data['status']` is `completed`. If we set the status to completed here then the `PAYMENT.CAPTURE.COMPLETED` event will be skipped without processing.

But we need to set some status here to mark that `capture_authorized_payment` function has already sent a request to capture the payment. This `capture_authorized_payment` method is called on order status change. And the order status change hook gets called more than once. If we do not check for a flag, there would be duplicate requests to capture the same payment. 

To avoid that I have added this custom status which is checked above at the beginning of the method.